synapse/rest/client/v1/login.py

```diff
@@ -298,11 +301,19 @@ async def _do_jwt_login(self, login_submission: JsonDict) -> Dict[str, str]:
             )
 
         import jwt
+        from jwt import PyJWKClient
+
+        key = self.jwt_secret
+
+        if not key and self.jwt_jwks_uri:
+            jwks_client = PyJWKClient(self.jwt_jwks_uri)
+            signing_key = jwks_client.get_signing_key_from_jwt(token)
```

Comment:
Does this make a synchronous HTTP call? If so we ideally would do this via a `SimpleHttpClient` or push this into the background.

Reply:
Fetching this for every login seems quite inefficient. I wonder if we should do something on start-up (like OIDC). It looks like `PyJWKClient` is quite simple!